modules/wallet/transactions.go

```diff
@@ -89,19 +104,27 @@ func (w *Wallet) Transactions(startHeight, endHeight types.BlockHeight) (pts []m
 	if startHeight > w.consensusSetHeight || startHeight > endHeight {
 		return nil, errOutOfBounds
 	}
-	if len(w.processedTransactions) == 0 {
-		return nil, nil
-	}
 
-	for _, pt := range w.processedTransactions {
-		if pt.ConfirmationHeight > endHeight {
-			break
-		}
-		if pt.ConfirmationHeight >= startHeight {
-			pts = append(pts, pt)
+	err = w.db.View(func(tx *bolt.Tx) error {
+		c := tx.Bucket(bucketProcessedTransactions).Cursor()
+		for key, val := c.First(); key != nil; key, val = c.Next() {
+			var pt modules.ProcessedTransaction
+			if err := encoding.Unmarshal(val, &pt); err != nil {
+				return err
+			}
+			if pt.ConfirmationHeight < startHeight {
+				continue
+			} else if pt.ConfirmationHeight > endHeight {
+				// transactions are stored in chronological order, so we can
+				// break as soon as we are above endHeight
```

Comment:
this basically makes the cacheing a requirement though. Doing 10,000 seeks + reads on a bolt database is going to take a while, especially on an HDD.